### ISSUE TITLE
TimeoutError: Pa11y timed out 

### ISSUE THREAD

**Issue by davidwilby** [pa11y/pa11y#662]
## Expected behaviour
run `pa11y http://localhost:4000/` for locally served website, get output

## Actual behaviour
```
pa11y -d http://localhost:4000

Welcome to Pa11y

 > Running Pa11y on URL http://localhost:4000
 > Debug: Launching Headless Chrome

Error: TimeoutError: Pa11y timed out (6000ms)
    at Timeout._onTimeout (/home/<user>/miniconda3/envs/<environment>/lib/node_modules/pa11y/node_modules/p-timeout/index.js:39:64)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
```

Note: pa11y is installed with npm in a conda environment

I've tried:
- increasing the timeout
- double-checking the locally running web server
- running against public websites
- Including `{ "chromeLaunchConfig": { "ignoreHTTPSErrors": false } }` in `./pa11y.json`
- an `nvm` install of node

All with the same results

## Steps to reproduce
1. Install nodejs in conda environment on ubuntu 22.04
2. `npm install pa11y`
3. `pa11y https://localhost:4000`

## Environment

```
  System:
    OS: Linux 5.17 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 2.36 GB / 15.27 GB
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.11.0 - ~/miniconda3/envs/<environment>/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.19.2 - ~/miniconda3/envs/<environment>bin/npm
  pa11y: 6.2.3
```

**Comment by davidwilby** [comment]
Seems like it might be an ubuntu 22.04 issue, see #651

**Comment by jankonzack** [comment]
Tried successfull running pa11y with a config setting "executablePath" within "chromeLaunchConfig" using Chromium (107.0.5304.87 snap).

`{
    "chromeLaunchConfig":
    {
        "ignoreHTTPSErrors": false,
        "executablePath": "/snap/bin/chromium"
    }
}`

`System:
    OS: Linux 5.19 Ubuntu 22.10 22.10 (Kinetic Kudu)
    CPU: (4) x64 Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
    Memory: 336.82 MB / 7.49 GB
    Shell: 5.2.2 - /bin/bash
  Binaries:
    Node: 15.4.0 - ~/.nvm/versions/node/v15.4.0/bin/node
    Yarn: 1.22.17 - /usr/bin/yarn
    npm: 7.0.15 - ~/.nvm/versions/node/v15.4.0/bin/npm
  pa11y: 6.2.3
`

**Comment by jankonzack** [comment]
As an addition: Problem seems to be an old version of puppeteer (in package.json: "puppeteer": "~9.1.1").
Installation of pa11y gave me warning about an old version of puppeteer.

Manually updating puppeteer (within the global pa11y installation directory) to latest solved the problem for me.

**Comment by tushar5526** [comment]
I faced a similar issue and @jankonzack solution works. Maybe you can make a PR for it.

**Comment by kussmaul** [comment]
As @jankonzack suggested, setting `executablePath` fixed a similar issue for me (using `pa11y-ci`). For me, (Fedora Core 36):
`"executablePath" : "/usr/lib64/chromium-browser/chromium-browser"`

It would be nice if the error messages had more information...

**Comment by GitarMan** [comment]
I had this same issue as well on Ubuntu 22.04. I was able to fix by adding a config file and setting the executablePath.

**Comment by danyalaytekin** [comment]
This was closed automatically by #664's merge but I don't think we're quite ready for that - reopening!
